packages/core/src/agent/openai/utils.ts

```diff
@@ -0,0 +1,35 @@
+import { ZodSchema } from "zod";
+import { ToolMetadata } from "../../types";
+import { getProperties } from "../utils";
+
+export type OpenAiFunction = {
```

Comment:
Let's just use the all uppercase `AI` in our class names. Yes, it's inconsistent with the way we do `Id` but too late/not enough reason to change now I think.